bdk_cli_lib/src/lib.rs

```diff
@@ -319,9 +319,7 @@ pub fn create_tx<P: ChainPosition>(
     };
 
     let (change_index, change_script) = {
-        let (index, script) = keychain_tracker
-            .txout_index
-            .derive_next_unused(&internal_keychain);
+        let (index, script) = keychain_tracker.txout_index.next_unused(&internal_keychain);
```

Comment:
I just realized that we have to persist the keychain index increment here!!!!!
I actually noticed this bug we were missing a change output in my local esplora example db.

I'll try and fix this tomorrow.